pyomo/gdp/plugins/cuttingplane.py

```diff
@@ -165,83 +784,247 @@ def _generate_cuttingplanes(
             raise GDP_Error("Cannot apply cutting planes transformation "
                             "without an active objective in the model!")
 
+        #
+        # Collect all of the linear constraints that are in the rBigM
+        # instance. We will need these so that we can compare what we get from
+        # FME to them and make sure we aren't adding redundant constraints to
+        # the model. For convenience, we will make sure they are all in the form
+        # lb <= expr (so we will break equality constraints)
+        #
+        fme = TransformationFactory('contrib.fourier_motzkin_elimination')
+        rBigM_linear_constraints = []
+        for cons in instance_rBigM.component_data_objects(
+                Constraint,
+                descend_into=Block,
+                sort=SortComponents.deterministic,
+                active=True):
+            body = cons.body
+            if body.polynomial_degree() != 1:
+                # We will never get a nonlinear constraint out of FME, so we
+                # don't risk it being identical to this one.
+                continue
+
+            # TODO: Guess this shouldn't have been private...
+            rBigM_linear_constraints.extend(fme._process_constraint(cons))
+
+        # [ESJ Aug 13 2020] NOTE: We actually don't need to worry about variable
+        # bounds here becuase the FME transformation will take care of them
+        # (i.e. convert those of the disaggregated variables to constraints for
+        # the purposes of the projection.)
+
+        return rBigM_obj, rBigM_linear_constraints
+
+    def _generate_cuttingplanes( self, instance_rBigM, instance_rHull, var_info,
+                                 transBlockName):
+
+        opt = SolverFactory(self._config.solver)
+        stream_solver = self._config.stream_solver
+        opt.options = self._config.solver_options
+
+        improving = True
+        prev_obj = float("inf")
```

Comment:
Wow, so there *are* actually maximization problems that are tested... It works in the code because I just check `math.isinf(obj_diff)` which is infinite in the first iteration either way. I'm using `float('inf')` as a convoluted placeholder for `None` I guess. Apparently `math.isinf(-float('inf'))` is `True` also (so the number line is a circle?!), so that means that for mathematical sanity, we could just set that to `-float('inf')` anyway. But perhaps it's clearer to just use `None` and never have to check the objective direction. I implemented that version.